certbot/plugins/selection.py

```diff
@@ -39,6 +39,35 @@ def pick_authenticator(
     return pick_plugin(
         config, default, plugins, question, (interfaces.IAuthenticator,))
 
+def get_installer(config, plugins):
+    """
+    Get an unprepared interfaces.IInstaller object.
+
+    :param certbot.interfaces.IConfig config: Configuration
+    :param certbot.plugins.disco.PluginsRegistry plugins:
+        All plugins registered as entry points.
+
+    :returns: Unprepared installer plugin.
+    :rtype: IPlugin
+    """
+
+    _, req_inst = cli_plugin_requests(config)
+    if not req_inst:
```

Comment:
We don't necessarily have to do things this way, but why not make an optional parameter to `pick_installer` and `pick_plugin` named something like `prepared`, add an `if` block around preparing the plugin in `pick_plugin`, and then just check that that function returns something other than `None`? We certainly don't need all of the logic in `pick_plugin`, but I think a lot of the code here is duplicated.

Another approach would be to split the common code in `pick_plugin` into helper functions both functions can use.

Reply:
This was the approach I initially took, it however became apparent, that `pick_plugin` itself is tightly knit into the interactive selection process, being the helper function that's called by other selection functions. It's bascally meant to be called mostly interactively - helping with the selection where needed, and I think that calling `prepare()` is a required part of the process with other verbs where interactive selection might be in question. Note that the `prepare()` is in some cases called to a whole group of applicable plugins where a single one wasn't requested. 

The current (semi-interactive) selection process is split between `choose_configurator_plugins` and `pick_plugin` that the former calls. This whole process is incorporated in this new function of this PR.

That said, I'd prefer to keep it as it is, as a robust selection process for interactive selection needs the `prepare()` called in the middle.

Reply:
That's fair. We're protected from any interactivity because `renew` implies `noninteractive` mode which `IDisplay` knows about, but there's enough differences in behavior it's probably not worth shoving it all into the same function. Thanks for pushing back.